drivers/flash/Kconfig.sam

```diff
@@ -7,6 +7,7 @@ config SOC_FLASH_SAM
 	bool "Atmel SAM flash driver"
 	default y
 	depends on DT_HAS_ATMEL_SAM_FLASH_CONTROLLER_ENABLED
+	depends on !SOC_SERIES_SAM3X
```

Comment:
if this doesn't work on sam3x why do we have the compatible in sam3x.dtsI?

Reply:
Because HWINFO requires `compatible = "atmel,sam-flash-controller";` to work.

https://github.com/zephyrproject-rtos/zephyr/blob/1c1801ee7ff3b36592a4804881cb78f23befb340/drivers/hwinfo/hwinfo_sam.c#L67

Is acceptable to keep `status = "disabled";` on this series to solve conflict?

I inspect more on sam4l too and it uses a different controller. If solution is not acceptable let me know what is best way to fix this with low effort right now. I don't have time to implement a driver from scratch right now.

Reply:
we should probably change the compatibles for flash, and update the hwinfo driver to support the different compatibles.